src/event/event_epoll.c

```diff
@@ -89,7 +89,7 @@ DISPATCH_ALWAYS_INLINE
 static inline uint32_t
 _dispatch_muxnote_armed_events(dispatch_muxnote_t dmn)
 {
-	return dmn->dmn_events & ~dmn->dmn_disarmed_events;
+	return dmn->dmn_events & ~(uint32_t)(dmn->dmn_disarmed_events);
```

Comment:
I think that casting to `uint16_t` is better as that is the declaration. That should avoid any unintentional zexting.

Reply:
The error says `'int' to 'uint32_t'`. I’m curious where the `int` comes from. If `int` is what the selected negation returns, then `(uint16_t)~dmn->dmn_disarmed_events` seems like it would cause a double conversion, whereas the current change appears to make it select a negation that returns an `uint32_t`.

Reply:
Switched to `decltype(dmn->dmn_events)` after discussing offline with Ben.